src/services/formatters/YscopeFormatter/index.ts

```diff
@@ -90,8 +92,16 @@ class YscopeFormatter implements Formatter {
                 throw Error("Field placeholder regex is invalid and does not have a capture group");
             }
 
-            const {parsedFieldName, formatterName, formatterOptions} =
-                splitFieldPlaceholder(groupMatch, this.#structuredIrNamespaceKeys);
+            const {fieldName, formatterName, formatterOptions} =
+                splitFieldPlaceholder(groupMatch);
+
+            const parsedFieldName: ParsedKey = parseKey(fieldName);
+            if (null === this.#structuredIrNamespaceKeys && parsedFieldName.isAutoGenerated) {
+                throw new Error(
+                    "`@` is a reserved symbol and must be escaped with `\\` " +
+                    "for JSONL logs."
```

Comment:
similar to the one thrown at `parseFilterKeys()`, i believe mentioning "CLP IR" helps with the context.

shall we consider making this message a const so it can be reused across multiple places?